pallene/checker.lua

```diff
@@ -135,22 +137,29 @@ end
 
 function Checker:add_global(decl)
     assert(decl._tag == "ast.Decl.Decl")
-    self.symbol_table:add_symbol(decl.name, checker.Name.Global(decl))
+    local name = decl._modname and (decl._modname .. '.' .. decl.name) or decl.name
+    local cname = self.symbol_table:find_symbol(name)
```

Comment:
This part where we  concatenate the modname caught my attention. My gut feeling is that instead of adding a `m.foo` item to the symbol table, the more natural thing to do would be to add the `m` module to the symbol table and then any use of `m.foo` would look for `foo` inside the module. The idea is that the name that's actually in scope is the `m`, not the `m.foo`. That said, if the current solution with the modname is working it's OK to keep it like this for now. I can always try changing it later if necessary. I'm just saying to keep an eye on this part of the design.

Reply:
Your idea is what I originally implemented, with the `fields` table being an element of the module symbol. The difference was that I inserted the `field` type in the `_name` value. I can try and go back to the previous version